### ISSUE TITLE
Wrap Markdown text in HTML tags

### ISSUE THREAD

**Issue by kaushalmodi** [russross/blackfriday#404]
Hello,

Is it possible to wrap Markdown text in HTML tags?

It will be cool if this [CommonMark-based solution presented in this SO answer][1] is feasible with BlackFriday.

## Use case

```
<div class="my_custom_table">

| a | b |
| c | d |

</div>
```

Then using [CSS table properties][2], I can customize that `.my_custom_table table` element to my heart's content.

[1]: https://stackoverflow.com/a/29378250/1219634
[2]: https://www.w3schools.com/css/css_table.asp

**Comment by kaushalmodi** [comment]
Probably related: https://github.com/russross/blackfriday/issues/364

**Comment by kaushalmodi** [comment]
Related: https://github.com/russross/blackfriday/issues/184

**Comment by kaushalmodi** [comment]
Another related one: https://github.com/russross/blackfriday/issues/181

**Comment by rtfb** [comment]
Isn't this the way Blackfriday currently behaves?

**Comment by kaushalmodi** [comment]
No, tried that exact snippet and it didn't work. I use Blackfriday through Hugo. Is there a way I can test if Blackfriday is doing things correctly?

**Comment by kaushalmodi** [comment]
@rtfb You can find the Markdown source [here][1], and the Hugo/Blackfriday rendered output [here][2].

[1]: https://gitlab.com/kaushalmodi/hugo-sandbox/raw/284a34d4a248820971226a11b8bedcbbee1f85e7/content/bugs/blackfriday-issue-404.md
[2]: https://hugo-sandbox.netlify.com/bugs/blackfriday-issue-404/

**Comment by shurcooL** [comment]
From https://github.com/russross/blackfriday/issues/184#issuecomment-121321333:

> This is expected behavior according to Markdown [spec](http://daringfireball.net/projects/markdown/syntax#html):
>
> > Note that Markdown formatting syntax is not processed within block-level HTML tags. E.g., you can’t use Markdown-style *emphasis* inside an HTML block.

**Comment by kaushalmodi** [comment]
@shurcooL Yes, I know that. I see that blackfriday incorporates "sensible" or user-friendly features that are not in the original Markdown spec.. take the table column alignment feature for example (using `:--`, `:--:` or `--:`).

So I am not claiming that it is not working as per the original Markdown spec.

I am rather proposing addition of this feature as per, apparently the CommanMark spec as per https://stackoverflow.com/questions/29368902/how-can-i-wrap-my-markdown-in-an-html-div/29378250#29378250 (as I say in my starting post of this issue).

**Comment by Titan-C** [comment]
After some hour of looking for an answer I resigned to get it working on blackfriday. I'm using a workaround in javascript now, just use something like this in your main HTML file or your JS file.

```javascript
<script type="text/javascript">
  document.getElementsByTagName("table")[0].classList.add("mytablecss");
</script>
```

This assumes you only have a table but it is no challenge to loop over all table elements and add the class desired for styling the table.

**Comment by kaushalmodi** [comment]
@Titan-C I believe you are using `ox-hugo`?

Read further if so :)

If using `ox-hugo`, I use a hack to wrap the Markdown tables in div's.

Here's a test: [Org][1] (search for "table-styling" in that Org file) | [Markdown][2] | [HTML][3]

[1]: https://raw.githubusercontent.com/kaushalmodi/ox-hugo/master/test/site/content-org/all-posts.org
[2]: https://raw.githubusercontent.com/kaushalmodi/ox-hugo/master/test/site/content/posts/table-styling.md
[3]: https://ox-hugo.scripter.co/test/posts/table-styling/

**Comment by Titan-C** [comment]
@kaushalmodi I don't know how I missed your documentation. My search engine was not personalized enough  I imagine. Thank you for your work

**Comment by kaushalmodi** [comment]
@Titan-C You are welcome.

Just FYI, I have site searches enabled:

- https://ox-hugo.scripter.co/search/?q=table
- https://ox-hugo.scripter.co/test/search/?q=table
